docs/blueprints/EP033.rst

```diff
@@ -0,0 +1,252 @@
+
+2) A shared **proxy port** for many UNIs (1:N).
+
+A share **proxy port** is a solution to address the need for full INT visibility when the amount of total traffic expected by all EVCs isn't higher than the capacity of the **proxy port**. This approach saves physical switch interfaces when users are not heavy users. For instance, consider a INT Sink switch where UNIs don't generate more than 10Gbps over a 100Gbps interface and there are four UNIs. Under normal circumstances, a 100Gbps **proxy port** should suffice for that INT Sink switch since we shouldn't see more than 40Gbps over that **proxy port**. Link aggregation can also be used to increase the **proxy port**'s capacity, for instance, to 200Gbps or more. A combination of #1 (Not using proxy ports) and #2 (Shared proxy port) can be implemented, where all inter-EVCs operate under the premises of #1 and intra-EVCs use the premises of #2.
+
+3) A dedicated **proxy port** per UNI (1:1).
+
+If one UNI along is responsible for 40% or higher of the capacity of a **proxy port**, EP033 suggests a dedicated **proxy port** for the offensive UNI. For instance, a heavy user could have a dedicated **proxy port** while non-heavy users can share a shared **proxy port** or mixing solutions #1, #2, and #3. Dedicated **proxy ports** are recommended for users considered high profile or burst traffic is expected.
+
+
+II. How to use proxy ports
+==========================
+
+As previously described, **proxy ports** are just physical loops placed by the network operators to re-inject packets into the pipeline for further processing. When **proxy ports** are used to enable telemetry, EVCs are broken down in two parts: (1) the source UNI to the **proxy port**, with support for telemetry, and (2) from the **proxy port** to the destination UNI without support for telemetry. The `Telemetry` napp will become responsible for addressing this new scenario as described in EP031.
+
+EP033 provides the following suggestion to integrate the approaches described in Section I with the `telemetry` napp:
+
+  * Leveraging the UNI's Interface object's `metadata` attribute to host a new key: "proxy_port". This new key can have the following values:
+
+    * 0 (Type Integer): When set to 0, any consumer of this attribute should assume that **proxy port** is not needed (Approach I.1). The same applies if there is not `proxy port` key in the metadata. If there is an intra-switch EVC and **proxy_port** is set to 0, there are two possibilities for the `telemetry` napp: use whatever **proxy port** is available or return to the operator that telemetry is not supported for such an EVC. EP031 is responsible for choosing the proper approach.
```

Comment:
Although this is not in the scope of this blueprint, you can iterate through all ports in the switch and search for the key proxy_port in the metadata. This activity will be performed by the telemetry napp and it will be defined in the EP031.

Reply:
A method can also be created in the Switch object for it or a metadata entry can be created in the Switch object's metadata listing all proxy ports. Suggestions?

Reply:
> Although this is not in the scope of this blueprint, you can iterate through all ports in the switch and search for the key proxy_port in the metadata. This activity will be performed by the telemetry napp and it will be defined in the EP031.

If I understood correctly, the proxy_port would be configured in the UNI, right? Assuming you are enabling INT in the first EVC and since this is the case of "when it is set to 0,or when there is no proxy port", it is still not clear to me how to list the available proxy ports.

Reply:
> A method can also be created in the Switch object for it or a metadata entry can be created in the Switch object's metadata listing all proxy ports. Suggestions?

I believe in this case, we would need additional metadata to be applied on the "proxy ports" (additional to the one applied on the UNI ports), to specify which ports are part of the "pool of proxy ports". wdyt?

Reply:
Iterating over `GET /topology/v3/interfaces` is fine, I think, plus if we can minimize round-trips that's even better. If we realize that it'll having to keep sending multiple request, instead of doing this for a group of EVCs, while still leveraging the metadata as it is strategically, then it could also subscribe to `kytos/topology.interfaces.metadata.(added|removed)` and `kytos/topology.topology_loaded` and keep it in sync in memory.

Reply:
Understood! Sounds good to me the way it is specified: it gives many possibilities for the Napps that will leverage this blueprint to implement any new service (e.g., the INT napp)